server/schedule/operator_controller.go

```diff
@@ -443,12 +444,20 @@ func (oc *OperatorController) addOperatorLocked(op *operator.Operator) bool {
 	operatorWaitDuration.WithLabelValues(op.Desc()).Observe(op.ElapsedTime().Seconds())
 	opInfluence := NewTotalOpInfluence([]*operator.Operator{op}, oc.cluster)
 	for storeID := range opInfluence.StoresInfluence {
-		stepCost := opInfluence.GetStoreInfluence(storeID).StepCost
-		if stepCost == 0 {
+		if oc.storesLimit[storeID] == nil {
 			continue
 		}
-		storeLimitGauge.WithLabelValues(strconv.FormatUint(storeID, 10), "take").Set(float64(stepCost) / float64(operator.RegionInfluence))
-		oc.storesLimit[storeID].Take(stepCost)
+		for n, v := range storelimit.TypeNameValue {
+			if oc.storesLimit[storeID][v] == nil {
+				continue
+			}
+			stepCost := opInfluence.GetStoreInfluence(storeID).GetStepCost(v)
```

Comment:
Assume that we can add 15 operators of balance region per minute, but now it must be less than 15 cause we take the remove peer cost into consideration, am I right?

Reply:
I depends the rate limit value of region-remove and region-add. If the value of region-remove is larger than region-add.  If no other region-remove operations，15 operators of balance region can still be added per minute.

Reply:
I depends the rate limit value of region-remove and region-add. If the value of region-remove is larger than region-add. If no other region-remove operations，15 operators of balance region can still be added per minute.

Reply:
But the removing peer operation will influence the speed of adding a peer since it may make the store unavailable. It's better to separate them?

Reply:
It is separated now. You can see `exceedStoreLimit` function.  If ops only have an adding peer operation, it won't be influenced.

Reply:
In `exceedStoreLimit`, if there are more than one operators in `ops`, they should be blocked or executed together.